packages/strapi-design-system/src/Toggle/Toggle.tsx

```diff
@@ -147,52 +149,28 @@ const Input = styled.input`
   width: 100%;
 `;
 
-interface ToggleInputPropsWithoutLabel
-  extends Pick<FieldProps, 'error' | 'hint' | 'name' | 'required' | 'id'>,
-    ToggleInputInputProps {
+interface TogglePropsWithoutLabel extends ToggleInputProps {
   clearLabel?: string;
   labelAction?: FieldLabelProps['action'];
   onClear?: () => void;
 }
 
-type ToggleInputProps =
-  | (ToggleInputPropsWithoutLabel & { label: string; 'aria-label'?: never })
-  | (ToggleInputPropsWithoutLabel & { label?: never; 'aria-label': string });
-
-const ToggleInput = React.forwardRef<ToggleInputInputElement, ToggleInputProps>(
-  (
-    {
-      disabled = false,
-      error,
-      hint,
-      label,
-      name,
-      labelAction,
-      required = false,
-      id,
-      onClear,
-      clearLabel,
-      checked,
-      ...props
-    },
-    forwardedRef,
-  ) => {
-    const generatedId = useId(id);
+type ToggleProps =
+  | (TogglePropsWithoutLabel & { label: string; 'aria-label'?: string })
+  | (TogglePropsWithoutLabel & { label?: never; 'aria-label': string });
 
+const Toggle = React.forwardRef<ToggleInputElement, ToggleProps>(
```

Comment:
This is a composite component no? Where we're still making this a whole field so it probably needs a rethink? 🤔

Reply:
yes, so you're saying? it should be called Input/Field?

Reply:
I'd see if we need it, can we just export the ToggleInput part as `Toggle`? Similar to how we do with the rest of the components :)